io_scene_dae/export_dae.py

```diff
@@ -1063,82 +1102,93 @@ def export_mesh_node(self, node, il):
             for sn in self.skeleton_info[armature]["skeleton_nodes"]:
-                self.writel(S_NODES, il + 1, '<skeleton>#' +
-                            sn + '</skeleton>')
+                self.writel(
+                    S_NODES, il + 1, "<skeleton>#{}</skeleton>".format(sn))
         elif ("morph_id" in meshdata):
-            self.writel(S_NODES, il, '<instance_controller url="#' +
-                        meshdata["morph_id"] + '">')
+            self.writel(
+                S_NODES, il, "<instance_controller url=\"#{}\">".format(
+                    meshdata["morph_id"]))
             close_controller = True
         elif (armature is None):
-            self.writel(S_NODES, il, '<instance_geometry url="#' +
-                        meshdata["id"] + '">')
+            self.writel(S_NODES, il, "<instance_geometry url=\"#{}\">".format(
+                meshdata["id"]))
 
         if (len(meshdata["material_assign"]) > 0):
 
-            self.writel(S_NODES, il + 1, '<bind_material>')
-            self.writel(S_NODES, il + 2, '<technique_common>')
+            self.writel(S_NODES, il + 1, "<bind_material>")
+            self.writel(S_NODES, il + 2, "<technique_common>")
             for m in meshdata["material_assign"]:
-                self.writel(S_NODES, il + 3, '<instance_material symbol="' +
-                            m[1] + '" target="#' + m[0] + '"/>')
+                self.writel(
+                    S_NODES, il + 3,
+                    "<instance_material symbol=\"{}\" target=\"#{}\"/>".format(
+                        m[1], m[0]))
 
-            self.writel(S_NODES, il + 2, '</technique_common>')
-            self.writel(S_NODES, il + 1, '</bind_material>')
+            self.writel(S_NODES, il + 2, "</technique_common>")
+            self.writel(S_NODES, il + 1, "</bind_material>")
 
         if (close_controller):
-            self.writel(S_NODES, il, '</instance_controller>')
+            self.writel(S_NODES, il, "</instance_controller>")
         else:
-            self.writel(S_NODES, il, '</instance_geometry>')
+            self.writel(S_NODES, il, "</instance_geometry>")
 
     def export_armature_bone(self, bone, il, si):
-        is_ctrl_bone = (bone.name.startswith("ctrl") and self.config["use_exclude_ctrl_bones"])
+        is_ctrl_bone = (
+            bone.name.startswith("ctrl") and
+            self.config["use_exclude_ctrl_bones"])
         if (bone.parent is None and is_ctrl_bone is True):
-            self.operator.report({'WARNING'},'Root bone cannot be a control bone.')
+            self.operator.report(
+                {"WARNING"}, "Root bone cannot be a control bone.")
             is_ctrl_bone = False
 
         if (is_ctrl_bone is False):
             boneid = self.new_id("bone")
             boneidx = si["bone_count"]
             si["bone_count"] += 1
-            bonesid = si["id"] + "-" + str(boneidx)
+            bonesid = + "{}-{}".format(si["id"], boneidx)
```

Comment:
I'm making a separate issue for the other ones but this needs to be fixed as part of this commit, otherwise the script won't run.